src/custom_nodes/common/utils.hpp

```diff
@@ -56,3 +63,419 @@ bool get_buffer(CustomNodeLibraryInternalManager* internalManager, T** buffer, c
     }
     return true;
 }
+
+template <typename T>
+void reorder_to_nhwc_2(const T* sourceNchwBuffer, T* destNhwcBuffer, int rows, int cols, int channels) {
+    for (int y = 0; y < rows; ++y) {
+        for (int x = 0; x < cols; ++x) {
+            for (int c = 0; c < channels; ++c) {
+                destNhwcBuffer[y * channels * cols + x * channels + c] = reinterpret_cast<const T*>(sourceNchwBuffer)[c * (rows * cols) + y * cols + x];
+            }
+        }
+    }
+}
+
+template <typename T>
+std::vector<T> reorder_to_nhwc(const T* nchwVector, int rows, int cols, int channels) {
+    std::vector<T> nhwcVector(rows * cols * channels);
+    reorder_to_nhwc_2(nchwVector, nhwcVector.data(), rows, cols, channels);
+    return nhwcVector;
+}
+
+template <typename T>
+void reorder_to_nchw_2(const T* sourceNhwcBuffer, T* destNchwBuffer, int rows, int cols, int channels) {
+    for (int y = 0; y < rows; ++y) {
+        for (int x = 0; x < cols; ++x) {
+            for (int c = 0; c < channels; ++c) {
+                destNchwBuffer[c * (rows * cols) + y * cols + x] = reinterpret_cast<const T*>(sourceNhwcBuffer)[y * channels * cols + x * channels + c];
+            }
+        }
+    }
+}
+
+template <typename T>
+std::vector<T> reorder_to_nchw(const T* nhwcVector, int rows, int cols, int channels) {
+    std::vector<T> nchwVector(rows * cols * channels);
+    reorder_to_nchw_2(nhwcVector, nchwVector.data(), rows, cols, channels);
+    return nchwVector;
+}
+
+const cv::Mat nhwc_to_mat(const CustomNodeTensor* input) {
+    uint64_t height = input->dims[1];
+    uint64_t width = input->dims[2];
+    return cv::Mat(height, width, CV_32FC3, input->data);
+}
+
+const cv::Mat nchw_to_mat(const CustomNodeTensor* input) {
+    uint64_t channels = input->dims[1];
+    uint64_t rows = input->dims[2];
+    uint64_t cols = input->dims[3];
+    auto nhwcVector = reorder_to_nhwc<float>((float*)input->data, rows, cols, channels);
+
+    cv::Mat image(rows, cols, CV_32FC3);
+    std::memcpy(image.data, nhwcVector.data(), nhwcVector.size() * sizeof(float));
+    return image;
+}
+
+bool crop_rotate_resize_limited(cv::Mat originalImage, cv::Mat& targetImage, cv::Rect roi, float angle, float originalTextWidth, float originalTextHeight, cv::Size targetShape) {
+    try {
+        // Limit roi to be in range of original image.
```

Comment:
Why do we need non limited roi? Opencv does not allow cropping image that go beyond original image. If we use non limited roi somewhere in custom nodes it means we potentially have bug.

Reply:
We were using non limited roi both in east_ocr and model_zoo_object_detection. crop_rotate_resize_limited(_limited was added by me) was originally in utilities of image_transformation